Clarify `get_translated_string` string argument (#13948)
minetest/minetest: doc/lua_api.md

a list on labels and send results to client. A method is supplied to achieve
that:

`minetest.get_translated_string(lang_code, string)`: resolves translations in
the given string just like the client would, using the translation files for
`lang_code`. For this to have any effect, the string needs to contain translation

> markup, e.g. `minetest.get_translated_string("fr", S("Hello")`.

Reviewer: ```suggestion
markup, e.g. `minetest.get_translated_string("fr", S("Hello"))`.
```

a list on labels and send results to client. A method is supplied to achieve
that:

> `minetest.get_translated_string(lang_code, S(string))`: resolves translations in

Reviewer 2: sorry, didn't see that before but: why is the `S` here?
the text already explains that the input is supposed to contain translations markup
and it is in fact correct to do e.g. `minetest.get_translated_string("", "foo\n" .. def.description .. "\nbar")`. No `S` in sight.

Author: I think people can still misunderstand such explanation. I've provided an example instead of listing the two translation methods, let me know

> the given string just like the client would, using the translation files for

Reviewer 3: I think you forgot the single quotes here: `string`

Reviewer: I'd say no, as string is a noun and not referencing verbatim code specifically

Reviewer 3: I guess then it has to say ”the given string `string`” or something like that?

Rationale: Without this patch the text used to reference the `string` argument and with this patch it does not.

`lang_code`. For this to have any effect, the string needs to contain translation

> markup (as made by `minetest.translate`, `get_translator`, or `S()`).

Reviewer 2: nitpick: `get_translator` does not return translation markup
and that's already covered by `S`, no? since that's the convention to use with get_translator

Author: ```suggestion
markup (as made by `minetest.translate` or `S()`).
```

a list on labels and send results to client. A method is supplied to achieve
that:

> `minetest.get_translated_string(lang_code, S(string))`: returns the translation in
> `lang_code` language of `S(string)`. It gives the same result as if the string was
> translated by the client.

Reviewer 2: my suggestion to keep it exact while still resolving the confusion:
```suggestion
`minetest.get_translated_string(lang_code, S(string))`: resolve translations in the given string
just like the client would, using the translation files for `lang_code`.
For this to have any effect the string has to contain what was returned by `minetest.translate`.
```

Author: Idk, I find mine more explicative but I'm not a native speaker. @rubenwardy ?

(also, *resolves)

Reviewer: sfan5's is more accurate, although I'd maybe change the last line to

> For this to have any effect, the string needs to contain translation markup (as made by `minetest.translate`, `get_translator`, or `S()`).

> `minetest.get_translated_string(lang_code, S(string))`: resolves translations in the given string
> just like the client would, using the translation files for `lang_code`.
> For this to have any effect, the string needs to contain translation markup (as made by `minetest.translate`, `get_translator`, or `S()`).

Reviewer 4: This text needs to be wrapped to account for the character limit per line (80 characters).
```suggestion
`minetest.get_translated_string(lang_code, S(string))`: resolves translations in
the given string just like the client would, using the translation files for
`lang_code`. For this to have any effect, the string needs to contain translation
markup (as made by `minetest.translate`, `get_translator`, or `S()`).
```
